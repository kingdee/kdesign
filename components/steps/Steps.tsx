import React, { useContext, cloneElement, ReactElement } from 'react'
import classNames from 'classnames'
import ConfigContext from '../config-provider/ConfigContext'
import { getCompProps } from '../_utils'
import devWarning from '../_utils/devwarning'
import { toArray } from '../_utils/react-children'
import { Icon } from '../index'
import { Directions, Direction, Statuses, Status, LabelPlacements, LabelPlacement, Icons } from './interface'
export interface StepsProps {
  direction?: Direction
  initial?: number
  current?: number
  icons?: Icons
  status?: Status
  className?: string
  style?: React.CSSProperties
  labelPlacement?: LabelPlacement
  onChange?: (current: number) => void
  canClickCurrentStep?: boolean
}

const Steps: React.FC<StepsProps> = (props) => {
  const { getPrefixCls, prefixCls, compDefaultProps: userDefaultProps } = useContext(ConfigContext)
  const stepsProps = getCompProps('Steps', userDefaultProps, props)
  const {
    direction,
    initial,
    current = 0,
    status = 'process',
    labelPlacement,
    prefixCls: customPrefixcls,
    className,
    style = {},
    onChange,
    children,
    icons,
    canClickCurrentStep,
    ...others
  } = stepsProps
  devWarning(Directions.indexOf(direction) === -1, 'steps', `cannot found steps direction '${direction}'`)
  devWarning(Statuses.indexOf(status) === -1, 'steps', `cannot found steps status '${status}'`)
  devWarning(
    LabelPlacements.indexOf(labelPlacement) === -1,
    'steps',
    `cannot found steps labelPlacement '${labelPlacement}'`,
  )
  const stepsPrefixCls = getPrefixCls!(prefixCls, 'steps', customPrefixcls) // 按钮样式前缀

  const stepsClassName = classNames(stepsPrefixCls, `${stepsPrefixCls}-${direction}`, className, {
    [`${stepsPrefixCls}-leftLable`]: labelPlacement === 'left',
    [`${stepsPrefixCls}-rightLable`]: labelPlacement === 'right',
    [`${stepsPrefixCls}-bottomLable`]: labelPlacement === 'bottom',
    [`${stepsPrefixCls}-topLable`]: labelPlacement === 'top',
  })

  const onStepClick = (next: number) => {
    if (canClickCurrentStep || (onChange && current !== next)) {
      onChange(next)
    }
  }

  const getIcons = () => {
    const actualIcons: Icons = {}
    actualIcons.finish = (icons && icons.finish) || <Icon type="right-bold" className={`${stepsPrefixCls}-iconSize`} />
    actualIcons.error = (icons && icons.error) || <Icon type="exclamatory" className={`${stepsPrefixCls}-iconSize`} />
    return actualIcons
  }

  const stepIcons = getIcons()

  return (
    <div className={stepsClassName} style={style} {...others}>
      {toArray(children).map((child: ReactElement, index: number) => {
        if (!child) return null
        const stepNumber = initial + index
        const childProps = {
          stepNumber: `${stepNumber + 1}`,
          stepIndex: stepNumber,
          key: stepNumber,
          prefixCls: stepsPrefixCls,
          icons: stepIcons,
          onStepClick: onChange && onStepClick,
          current,
          ...child.props,
        }
        if (!child.props.status) {
          if (stepNumber === current) {
            childProps.status = status
          } else if (stepNumber < current) {
            childProps.status = 'finish'
          } else {
            childProps.status = 'wait'
          }
        }
        return cloneElement(child, childProps)
      })}
    </div>
  )
}

Steps.displayName = 'Steps'
export default Steps
