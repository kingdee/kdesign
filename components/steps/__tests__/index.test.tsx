import React from 'react'
import { render, mount } from 'enzyme'
import ConfigProvider from '../../config-provider/index'
import Icon from '../../icon'
import Steps, { Step, Directions, Direction, LabelPlacement, LabelPlacements, Statuses, Status } from '../index'
import mountTest from '../../../tests/shared/mountTest'

describe('Steps', () => {
  // 1.mount test
  describe('1.mount test', () => {
    mountTest(Steps)
    Directions.forEach((direction) => {
      mountTest(() => (
        <Steps direction={direction}>
          <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
          <Step title="Step2" description="hello word" />
        </Steps>
      ))
    })
  })

  // 2.render test
  describe('2.render test', () => {
    it('renders correctly', () => {
      expect(
        render(
          <Steps>
            <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
            <Step title="Step2" description="hello word" />
          </Steps>,
        ),
      ).toMatchSnapshot()

      Directions.forEach((direction) => {
        expect(
          render(
            <Steps direction={direction}>
              <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
              <Step title="Step2" description="hello word" />
            </Steps>,
          ),
        ).toMatchSnapshot()
      })
      LabelPlacements.forEach((labelPlacement) => {
        expect(
          render(
            <Steps labelPlacement={labelPlacement}>
              <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
              <Step title="Step2" description="hello word" />
            </Steps>,
          ),
        ).toMatchSnapshot()
      })
      Statuses.forEach((status) => {
        expect(
          render(
            <Steps status={status}>
              <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
              <Step title="Step2" description="hello word" />
            </Steps>,
          ),
        ).toMatchSnapshot()
      })

      const icons = {
        process: <Icon type="loadding" spin />,
        wait: <Icon type="wating" />,
        error: <Icon type="close" />,
        finish: <Icon type="right-bold" />,
      }

      expect(
        render(
          <Steps icons={icons} current={2} status="error">
            <Step title="Step1" description="hello word" icon={<Icon type="right-bold" />} />
            <Step title="Step2" description="hello word" />
            <Step title="Step3" description="hello word" />
            <Step title="Step4" description="hello word" />
          </Steps>,
        ),
      ).toMatchSnapshot()
    })
  })

  // 3.warns in component
  describe('3.warns in component', () => {
    it('warns if direction is wrong', () => {
      const mockWarn = jest.fn()
      jest.spyOn(console, 'warn').mockImplementation(mockWarn)
      const direction = 'who am I' as any as Direction
      render(<Steps direction={direction} />)
      expect(mockWarn).toHaveBeenCalledTimes(1)
      expect(mockWarn.mock.calls[0][0]).toMatch("Warning: [kdesign]-steps: cannot found steps direction 'who am I'")
    })

    it('warns if labelPlacement is wrong', () => {
      const mockWarn = jest.fn()
      jest.spyOn(console, 'warn').mockImplementation(mockWarn)
      const labelPlacement = 'who am I' as any as LabelPlacement
      render(<Steps labelPlacement={labelPlacement} />)
      expect(mockWarn).toHaveBeenCalledTimes(1)
      expect(mockWarn.mock.calls[0][0]).toMatch(
        "Warning: [kdesign]-steps: cannot found steps labelPlacement 'who am I'",
      )
    })
    it('warns if status is wrong', () => {
      const mockWarn = jest.fn()
      jest.spyOn(console, 'warn').mockImplementation(mockWarn)
      const status = 'who am I' as any as Status
      render(<Steps status={status} />)
      expect(mockWarn).toHaveBeenCalledTimes(1)
      expect(mockWarn.mock.calls[0][0]).toMatch("Warning: [kdesign]-steps: cannot found steps status 'who am I'")
    })
  })

  // 4. render null or undefined without errors
  describe('4.render null or undefined without errors', () => {
    it('render null or undefined buttons without errors', () => {
      expect(
        mount(
          <Steps>
            {null}
            {undefined}
          </Steps>,
        ),
      ).toMatchSnapshot()
    })
  })

  // 5. displayName
  describe('5.displayName', () => {
    it('should have displayName static property', () => {
      const wrapper = mount(
        <Steps>
          <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
          <Step title="Step2" description="hello word" />
        </Steps>,
      )
      expect((wrapper.type() as any).displayName).toBe('Steps')
    })
  })

  // 6. class state
  describe('6.class state', () => {
    it('should class use right', () => {
      // default
      const DefaultSteps = mount(
        <Steps>
          <Step title="Step1" description="hello word hello word hello word hello word hello word hello word" />
          <Step title="Step2" description="hello word" />
        </Steps>,
      )
      expect(DefaultSteps.find('.kd-steps')).toHaveClassName('.kd-steps-horizontal')
      expect(DefaultSteps.find('.kd-steps')).toHaveClassName('.kd-steps-bottomLable')
      const mockOnChange = jest.fn()
      const clickableSteps = mount(
        <Steps onChange={mockOnChange}>
          <Step title="Step1" description="hello word" />
        </Steps>,
      )
      expect(clickableSteps.find('.kd-steps-item')).toHaveClassName('.kd-steps-item-clickable')

      Directions.forEach((direction) => {
        const TestSteps = mount(
          <Steps direction={direction}>
            <Step title="Step1" description="hello word" />
          </Steps>,
        )
        expect(TestSteps.find(`.kd-steps`)).toHaveClassName(`.kd-steps-${direction}`)
      })

      LabelPlacements.forEach((labelPlacement) => {
        const TestSteps = mount(
          <Steps labelPlacement={labelPlacement}>
            <Step title="Step1" description="hello word" />
          </Steps>,
        )
        expect(TestSteps.find(`.kd-steps`)).toHaveClassName(`.kd-steps-${labelPlacement}Lable`)
      })
      Statuses.forEach((status) => {
        const TestSteps = mount(
          <Steps status={status}>
            <Step title="Step1" description="hello word" />
          </Steps>,
        )
        expect(TestSteps.find(`.kd-steps-item`)).toHaveClassName(`.kd-steps-item-${status}`)
      })
    })
  })
  // 7.component interaction(event)
  describe('7.component interaction(event)', () => {
    it('should clickable ', () => {
      const onChange = jest.fn()
      const wrapper = mount(
        <Steps onChange={onChange}>
          <Step title="Step1" description="hello word" />
          <Step title="Step2" description="hello word" />
          <Step title="Step3" description="hello word" />
        </Steps>,
      )
      wrapper.find('.kd-steps-item').at(1).simulate('click')
      expect(onChange).toHaveBeenCalled()
    })
  })

  // 8.config provider
  describe('8.config provider', () => {
    it('should config use config provider', () => {
      const stepsConfig = {
        compDefaultProps: {
          Steps: {
            direction: 'vertical',
          },
        },
      }
      const wrapper = mount(
        <ConfigProvider value={stepsConfig}>
          <Steps>
            <Step title="Step1" description="hello word" />
          </Steps>
        </ConfigProvider>,
      )
      expect(wrapper.find('.kd-steps')).toHaveClassName('.kd-steps-vertical')
    })
  })

  // 9.API test
  describe('9.api test', () => {
    it('Steps API test', () => {
      // canClickCurrentStep
      const onChange = jest.fn()
      const canClickCurrentSteps = mount(
        <Steps onChange={onChange} canClickCurrentStep={true}>
          <Step title="Step1" description="hello word" />
          <Step title="Step2" description="hello word" />
          <Step title="Step3" description="hello word" />
        </Steps>,
      )
      canClickCurrentSteps.find('.kd-steps-item').at(0).simulate('click')
      expect(onChange).toHaveBeenCalled()
      // current
      const currentOnChange = jest.fn()
      const currentSteps = mount(
        <Steps onChange={currentOnChange} current={1}>
          <Step title="Step1" description="hello word" />
          <Step title="Step2" description="hello word" />
          <Step title="Step3" description="hello word" />
        </Steps>,
      )
      currentSteps.find('.kd-steps-item').at(1).simulate('click')
      expect(currentOnChange).not.toHaveBeenCalled()
      // initial
      const initialSteps = mount(
        <Steps initial={2}>
          <Step title="步骤条节点标题文本，我是比较长的文本" />
          <Step title="标题内容2" />
          <Step title="标题内容3" />
          <Step title="标题内容4" />
        </Steps>,
      )
      expect(initialSteps.find('.kd-steps-item').at(0).find('.kd-steps-icon').text()).toEqual(`3`)
    })
    it('Steps.step API test', () => {
      // status
      const statusSteps = mount(
        <Steps>
          <Step title="标题内容1" status="error" />
          <Step title="标题内容2" status="wait" />
          <Step title="标题内容3" status="process" />
          <Step title="标题内容4" status="finish" />
        </Steps>,
      )
      expect(statusSteps.exists('.kd-steps-item-error')).toBeTruthy()
      expect(statusSteps.exists('.kd-steps-item-wait')).toBeTruthy()
      expect(statusSteps.exists('.kd-steps-item-process')).toBeTruthy()
      expect(statusSteps.exists('.kd-steps-item-finish')).toBeTruthy()
      // stepNumber
      const stepNumberSteps = mount(
        <Steps>
          <Step title="标题内容1" stepNumber={200} />
        </Steps>,
      )
      expect(stepNumberSteps.find('.kd-steps-item').at(0).find('.kd-steps-icon').text()).toEqual(`200`)
    })
  })
})
