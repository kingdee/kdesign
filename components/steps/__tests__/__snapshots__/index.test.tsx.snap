// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps 2.render test renders correctly 1`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 2`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 3`] = `
<div
  class="kd-steps kd-steps-vertical kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 4`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-leftLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 5`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-rightLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 6`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-topLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 7`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 8`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 9`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-process"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            1
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 10`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-finish"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            <i
              class="kdicon kd-steps-iconSize kdicon-right-bold"
            />
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 11`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-error"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            <i
              class="kdicon kd-steps-iconSize kdicon-exclamatory"
            />
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word hello word hello word hello word hello word hello word"
        >
          hello word hello word hello word hello word hello word hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 2.render test renders correctly 12`] = `
<div
  class="kd-steps kd-steps-horizontal kd-steps-bottomLable"
>
  <div
    class="kd-steps-item kd-steps-item-finish"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer kd-steps-item-iconContainer-finish"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            <i
              class="kdicon kdicon-right-bold"
            />
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step1"
        >
          Step1
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-finish"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer kd-steps-item-iconContainer-finish"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            <i
              class="kdicon kdicon-right-bold"
            />
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step2"
        >
          Step2
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-error"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            <i
              class="kdicon kdicon-close"
            />
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step3"
        >
          Step3
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
  <div
    class="kd-steps-item kd-steps-item-wait"
  >
    <div
      class="kd-steps-item-container"
    >
      <div
        class="kd-steps-item-iconContainer"
      >
        <div
          class="kd-steps-item-iconContainer-icon"
        >
          <span
            class="kd-steps-icon"
          >
            4
          </span>
        </div>
      </div>
      <div
        class="kd-steps-item-content"
      >
        <div
          class="kd-steps-item-title"
          title="Step4"
        >
          Step4
        </div>
        <div
          class="kd-steps-item-description"
          title="hello word"
        >
          hello word
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps 4.render null or undefined without errors render null or undefined buttons without errors 1`] = `
<Steps>
  <div
    className="kd-steps kd-steps-horizontal kd-steps-bottomLable"
    style={Object {}}
  />
</Steps>
`;
